src/mito2/src/sst/index/creator/temp_provider.rs

```diff
@@ -0,0 +1,172 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use async_trait::async_trait;
+use common_error::ext::BoxedError;
+use common_telemetry::warn;
+use futures::{AsyncRead, AsyncWrite};
+use index::inverted_index::create::sort::external_provider::ExternalTempFileProvider;
+use index::inverted_index::error as index_error;
+use index::inverted_index::error::Result as IndexResult;
+use snafu::ResultExt;
+
+use crate::error::Result;
+use crate::metrics::{
+    INDEX_INTERMEDIATE_FLUSH_OP_TOTAL, INDEX_INTERMEDIATE_READ_BYTES_TOTAL,
+    INDEX_INTERMEDIATE_READ_OP_TOTAL, INDEX_INTERMEDIATE_SEEK_OP_TOTAL,
+    INDEX_INTERMEDIATE_WRITE_BYTES_TOTAL, INDEX_INTERMEDIATE_WRITE_OP_TOTAL,
+};
+use crate::sst::index::store::InstrumentedStore;
+use crate::sst::location::IntermediateLocation;
+
+/// `TempFileProvider` implements `ExternalTempFileProvider`.
+/// It uses `InstrumentedStore` to create and read intermediate files.
+pub(crate) struct TempFileProvider {
+    /// Provides the location of intermediate files.
+    location: IntermediateLocation,
+    /// Provides access to files in the object store.
+    store: InstrumentedStore,
+}
+
+#[async_trait]
+impl ExternalTempFileProvider for TempFileProvider {
+    async fn create(
+        &self,
+        column_name: &str,
```

Comment:
Which one is better, column name or column id? We are working at the storage engine layer, where column id is used wider than column name. If this index creating procedure will be affected by DDL, or run across different versions of schema, the column name is not applicable, right?